app/browser/tabs.js

```diff
@@ -1019,9 +1021,36 @@ const api = {
           createProperties.parent_partition = ''
         }
         if (createProperties.isTor) {
+          // TODO(riastradh): Duplicate logic in app/filtering.js.
           createProperties.isolated_storage = true
           createProperties.parent_partition = ''
-          createProperties.tor_proxy = 'socks5://127.0.0.1:9050'
+          let portno
+          switch (channel()) {
+            case 'dev':
+            case '':
+            default:
+              portno = 9250
+              break
+            case 'beta':
+              portno = 9260
+              break
+            case 'nightly':
+              portno = 9270
+              break
+            case 'developer':
+              portno = 9280
+              break
+          }
+          console.log(`channel ${channel()} tor port ${portno}`)
+          createProperties.tor_proxy = `socks5://127.0.0.1:${portno}`
+          const etcDir = getExtensionsPath('etc')
+          try {
+            fs.mkdirSync(path.join(etcDir, 'tor'), 0o0700)
```

Comment:
we shouldn't call synchronous fs methods for performance reasons. can you just make this directory and check it into git? i would prefer it be `app/extensions/brave/etc/tor` since Tor is part of the Brave "extension" mostly